cmd/sanitize-prow-jobs/main.go

```diff
@@ -97,28 +138,64 @@ func determinizeJobs(prowJobConfigDir string) error {
 	return utilerrors.NewAggregate(errs)
 }
 
-func defaultJobConfig(jc *prowconfig.JobConfig) {
+func defaultJobConfig(jc *prowconfig.JobConfig, path string, config *Config) {
 	for k := range jc.PresubmitsStatic {
 		for idx := range jc.PresubmitsStatic[k] {
-			defaultJobBase(&jc.PresubmitsStatic[k][idx].JobBase)
+			defaultJobBase(&jc.PresubmitsStatic[k][idx].JobBase, path, config)
 		}
 	}
 	for k := range jc.PostsubmitsStatic {
 		for idx := range jc.PostsubmitsStatic[k] {
-			defaultJobBase(&jc.PostsubmitsStatic[k][idx].JobBase)
+			defaultJobBase(&jc.PostsubmitsStatic[k][idx].JobBase, path, config)
 		}
 	}
 	for idx := range jc.Periodics {
-		defaultJobBase(&jc.Periodics[idx].JobBase)
+		defaultJobBase(&jc.Periodics[idx].JobBase, path, config)
 	}
 }
 
-func defaultJobBase(jb *prowconfig.JobBase) {
+func defaultJobBase(jb *prowconfig.JobBase, path string, config *Config) {
+	jb.Cluster = getCluster(jb.Name, path, config)
+}
 
-	if jb.Cluster == "" || jb.Cluster == "default" {
-		jb.Cluster = defaultCluster
+func getCluster(name string, path string, config *Config) string {
```

Comment:
Please rename `getCluster` -> `getClusterForJob` and `name` -> `jobName`.

Would it maybe make sense to make `getCluster` a method of `config` to have them closer together and signal the tight coupling?

Reply:
Done.